src/util/MolgenisFormGenerator.js

```diff
@@ -251,25 +251,36 @@ const generateFormSchemaField = (attribute): FormField => {
  *
  * @param fields an array of field objects
  * @param data a data object containing everything a EntityType V2 response has in its item list
+ * @param attributes an array of MOLGENIS attribute metadata, used for idAttribute
  * @returns a {fieldId: value} object
  */
-const generateFormData = (fields: any, data: any) => {
+const generateFormData = (fields: any, data: any, attributes: any) => {
   return fields.reduce((accumulator, field) => {
-    if (field.type === 'field-group') {
-      return {...accumulator, ...generateFormData(field.children, data)}
-    } else if (field.type === 'file') {
-      // Map MOLGENIS FileMeta entity to our form file object
-      // which only contains a name
-      const fileData = data[field.id]
-      accumulator[field.id] = fileData ? fileData.filename : data[field.id]
-    } else if (field.type === 'checkbox' || field.type === 'multi-select') {
-      const checkboxData = data[field.id]
-      accumulator[field.id] = checkboxData && checkboxData.map(data => data.id)
-    } else if (field.type === 'radio' || field.type === 'single-select') {
-      const radioData = data[field.id]
-      accumulator[field.id] = radioData && typeof radioData === 'object' ? radioData.id : data[field.id]
-    } else {
-      accumulator[field.id] = data[field.id]
+    const fieldAttribute = attributes.find(attribute => attribute.name === field.id)
```

Comment:
why combine the move from if else to switch with the bug fix , this makes it hard to review the PR